### DISCUSSION TITLE
Allow changing default configuration values of the extension

### DISCUSSION THREAD

**Discussion by synulux** [rome/tools#4603] [Suggestions]
It would be nice to be able to change the default configuration values of the extension, so that when the `rome.requireConfiguration` option is `false`, we can specify ourselves how we want the extension's linter & formatter to behave.

Example:
```jsonc
// settings.json (Visual Studio Code)
{
  "rome.requireConfiguration": false,

  "rome.configuration": {
    "formatter": {
      "enabled": true
    },
    "linter": {
      "enabled": true,
      "rules": {
        "recommended": true
      }
    },
    "organizeImports": {
      "enabled": true
    }
  }
}
```

Unsure if this is already possible, but if so, let me know!

**Comment by ematipico** [comment]
It **was** possible at the beginning when we had just the formatter and we didn't have a configuration file. We removed it once we started to have more things.

The issue we saw was were the following:
- the number of options
- which options
- what about cases where the user had the configuration file too? That's a concern for CI environments that we can't overlook

Do you have some idea?

**Reply by synulux** [reply]
For this one: `what about cases where the user had the configuration file too? That's a concern for CI environments that we can't overlook`, I believe it's possible to do the same thing Prettier does with their extension [(click here for more)](https://github.com/prettier/prettier-vscode#configuration):

- **`rome.json` file is not present in the current project, and `%AppData%/Code/User/settings.json > rome.configuration` is not present aswell:** The Rome extension would automatically use the default configuration values.
-  **`rome.json` file is not present in the current project, but `%AppData%/Code/User/settings.json > rome.configuration` is present:** The Rome extension would not use the default configuration, but the configuration that the user added in the Visual Studio Code (or any other editor) settings.
- **`rome.json` file is not present in the current project, but `myProject/.vscode/settings.json > rome.configuration` is present:** Even if `%AppData%/Code/User/settings.json > rome.configuration` is present, the Rome extension would ignore it and use the `myProject/.vscode/settings.json > rome.configuration` values.
- **`rome.json` file is present in the current project:** The Rome extension would use this configuration file even if `%AppData%/Code/User/settings.json > rome.configuration` or `myProject/.vscode/settings.json > rome.configuration` is present.
